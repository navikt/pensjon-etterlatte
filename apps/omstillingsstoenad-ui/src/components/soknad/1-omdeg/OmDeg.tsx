import SoknadSteg from '../../../typer/SoknadSteg'
import { useTranslation } from 'react-i18next'
import InnloggetBruker from './InnloggetBruker'
import { SkjemaGruppe } from '../../felles/SkjemaGruppe'
import { FormProvider, useForm } from 'react-hook-form'
import { IValg } from '../../../typer/Spoersmaal'
import { useSoknadContext } from '../../../context/soknad/SoknadContext'
import { ISoeker } from '../../../typer/person'
import { ActionTypes } from '../../../context/soknad/soknad'
import { RHFInput, RHFKontonummerInput, RHFTelefonInput } from '../../felles/rhf/RHFInput'
import { RHFInlineRadio, RHFSpoersmaalRadio } from '../../felles/rhf/RHFRadio'
import Feilmeldinger from '../../felles/Feilmeldinger'
import { useBrukerContext } from '../../../context/bruker/BrukerContext'
import Navigasjon from '../../felles/Navigasjon'
import { Cell, Grid, Heading, RadioProps } from '@navikt/ds-react'
import { BankkontoType } from '../../../typer/utbetaling'
import UtenlandskBankInfo from './utenlandskBankInfo/UtenlandskBankInfo'
import HvorforSpoerVi from '../../felles/HvorforSpoerVi'
import { deepCopy } from '../../../utils/deepCopy'
import { RHFSelect } from '../../felles/rhf/RHFSelect'
import { useLand } from '../../../hooks/useLand'
import { SkjemaElement } from '../../felles/SkjemaElement'
import Bredde from '../../../typer/bredde'

const OmDeg: SoknadSteg = ({ neste }) => {
    const { t } = useTranslation()
    const { state, dispatch } = useSoknadContext()
    const brukerState = useBrukerContext().state
    const { land }: { land: any } = useLand()
    const lagre = (data: ISoeker) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG, payload: { ...deepCopy(data), erValidert: true } })
        neste!!()
    }

    const methods = useForm<ISoeker>({
        defaultValues: state.omDeg || {},
        shouldUnregister: true,
    })

    const {
        handleSubmit,
        watch,
        formState: { errors },
    } = methods

    const skalSjekkeFlyktningStatus = brukerState.foedselsaar!! < 1960

    const oppholderSegINorge = watch('oppholderSegINorge')
    const bankkontoType = watch('utbetalingsInformasjon.bankkontoType')

    return (
        <>
            <SkjemaElement>
                <Heading size={'medium'} className={'center'}>
                    {t('omDeg.tittel')}
                </Heading>
            </SkjemaElement>
            <InnloggetBruker />

            <FormProvider {...methods}>
                {/* TODO: Flytte dette til start eller eget steg? */}

                <form>
                    <SkjemaGruppe>
                        {!brukerState.adressebeskyttelse && !brukerState.adresse && (
                            <SkjemaGruppe>
                                <RHFInput name={'alternativAdresse'} label={t('omDeg.alternativAdresse')} />
                            </SkjemaGruppe>
                        )}

                        {!brukerState.telefonnummer && (
                            <SkjemaGruppe>
                                <Grid>
                                    <Cell xs={12} md={6} className={'kol'}>
                                        <RHFTelefonInput
                                            htmlSize={Bredde.S}
                                            name={'kontaktinfo.telefonnummer'}
                                            label={t('omDeg.kontaktinfo.telefonnummer')}
                                            valgfri={true}
                                        />
                                    </Cell>
                                </Grid>
                            </SkjemaGruppe>
                        )}
                    </SkjemaGruppe>

                    {/* 2.7 */}
                    {!brukerState.adressebeskyttelse && (
                        <SkjemaGruppe>
                            <SkjemaElement>
                                <RHFSpoersmaalRadio
                                    name={'oppholderSegINorge'}
                                    legend={t('omDeg.oppholderSegINorge')}
                                    description={
                                        <HvorforSpoerVi title="omDeg.oppholderSegINorge">
                                            {t('omDeg.oppholdHvorfor')}
                                        </HvorforSpoerVi>
                                    }
                                />
                            </SkjemaElement>

                            {oppholderSegINorge === IValg.JA && (
                                <SkjemaGruppe>
                                    <RHFKontonummerInput
                                        htmlSize={Bredde.S}
                                        name={'utbetalingsInformasjon.kontonummer'}
                                        label={t('omDeg.utbetalingsInformasjon.kontonummer')}
                                        placeholder={t('felles.elleveSiffer')}
                                        description={t('omDeg.utbetalingsInformasjon.informasjon')}
                                    />
                                </SkjemaGruppe>
                            )}

                            {oppholderSegINorge === IValg.NEI && (
                                <>
                                    <SkjemaGruppe>
                                        <RHFSelect
                                            className="kol-50"
                                            name={`oppholdsland`}
                                            label={t('omDeg.oppholdsland')}
                                            selectOptions={land}
                                        />
                                    </SkjemaGruppe>

                                    <SkjemaElement>
                                        <RHFInlineRadio
                                            name={'utbetalingsInformasjon.bankkontoType'}
                                            legend={t('omDeg.utbetalingsInformasjon.bankkontoType')}
                                        >
                                            {Object.values(BankkontoType).map((value) => {
                                                return { children: t(value), value } as RadioProps
                                            })}
                                        </RHFInlineRadio>
                                    </SkjemaElement>

                                    {bankkontoType === BankkontoType.norsk && (
                                        <RHFKontonummerInput
                                            htmlSize={Bredde.S}
                                            name={'utbetalingsInformasjon.kontonummer'}
                                            label={t('omDeg.utbetalingsInformasjon.kontonummer')}
                                            placeholder={t('felles.elleveSiffer')}
                                        />
                                    )}

                                    {bankkontoType === BankkontoType.utenlandsk && <UtenlandskBankInfo />}
                                </>
                            )}
                        </SkjemaGruppe>
                    )}

                    {skalSjekkeFlyktningStatus && (
                        <SkjemaGruppe>
                            <RHFSpoersmaalRadio
                                name={'flyktning'}
                                legend={t('omDeg.flyktning')}
                                description={
                                    <HvorforSpoerVi title="omDeg.flyktning">
                                        {t('omDeg.flyktningHvorfor')}
                                    </HvorforSpoerVi>
                                }
                            />
                        </SkjemaGruppe>
                    )}

                    <br />

                    <Feilmeldinger errors={errors} />

                    <Navigasjon neste={{ onClick: handleSubmit(lagre) }} />
                </form>
            </FormProvider>
        </>
    )
}

export default OmDeg
