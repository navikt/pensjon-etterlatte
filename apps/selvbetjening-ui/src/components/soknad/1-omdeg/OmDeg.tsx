import "./OmDeg.less";
import { Systemtittel } from "nav-frontend-typografi";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useTranslation } from "react-i18next";
import InnloggetBruker from "./InnloggetBruker";
import { Hovedknapp } from "nav-frontend-knapper";
import { SkjemaGruppe } from "nav-frontend-skjema";
import { FormProvider, useForm } from "react-hook-form";
import { IValg } from "../../../typer/Spoersmaal";
import AlertStripe from "nav-frontend-alertstriper";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { ISoeker } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { RHFInput, RHFKontonummerInput, RHFTelefonInput } from "../../felles/RHFInput";
import { RHFToValgRadio } from "../../felles/RHFRadio";
import Feilmeldinger from "../../felles/Feilmeldinger";
import { useBrukerContext } from "../../../context/bruker/BrukerContext";
import React from "react";

const OmDeg: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();
    const brukerState = useBrukerContext().state;

    const lagre = (data: ISoeker) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG, payload: data })
        neste!!()
    };

    const methods = useForm<ISoeker>({
        defaultValues: state.omDeg || {},
        shouldUnregister: true
    });

    const {
        handleSubmit,
        watch,
        formState: { errors }
    } = methods;

    const skalSjekkeFlyktningStatus = brukerState.foedselsaar!! < 1960;

    const borPaaRegistrertAdresse = watch("bostedsadresseBekreftet")
    const oppholderSegINorge = watch("oppholderSegINorge")

    return (
        <>
            {/* Steg 2 */}
            <Systemtittel className={"center"}>
                {t("omSoekeren.tittel")}
            </Systemtittel>

            {/* Informasjon om den innloggede brukeren */}
            <InnloggetBruker />

            {/* Skjema for utfylling av info om innlogget bruker / søker */}
            <FormProvider {...methods}>
                {/* TODO: Flytte dette til start eller eget steg? */}

                <form>
                    <RHFToValgRadio
                        name={"bostedsadresseBekreftet"}
                        legend={t("omSoekeren.borPaaAdresse")}
                    />

                    {borPaaRegistrertAdresse === IValg.NEI && (
                        <SkjemaGruppe>
                            <AlertStripe type="advarsel">{t("omSoekeren.infoFolkeregisteret")}</AlertStripe>
                        </SkjemaGruppe>
                    )}

                    <div className={"rad skjemagruppe"}>
                        <div className={"kol-50"}>
                            <RHFTelefonInput
                                name={"kontaktinfo.telefonnummer"}
                                label={t("omSoekeren.kontaktinfo.telefon")}
                                // TODO: Validere telefon ... ?
                            />
                        </div>

                        <div className={"kol-50"}>
                            {/* 2.5 */}
                            <RHFInput
                                name={"kontaktinfo.epost"}
                                label={t("omSoekeren.kontaktinfo.epost")}
                                // TODO: Validere e-post
                            />
                        </div>
                    </div>

                    {/* 2.7 */}
                    <RHFToValgRadio
                        name={"oppholderSegINorge"}
                        legend={t("omSoekeren.oppholderSegINorge")}
                    />

                    {oppholderSegINorge === IValg.JA && (
                        <SkjemaGruppe>
                            <RHFKontonummerInput
                                name={"kontonummer"}
                                label={t("Oppgi norsk kontonummer for utbetaling")}
                                rules={{pattern: /^\d{4}\.\d{2}\.\d{5}$/}}
                                placeholder={"11 siffer"}
                            />
                        </SkjemaGruppe>
                    )}

                    {oppholderSegINorge === IValg.NEI && (
                        <>
                            <SkjemaGruppe>
                                <RHFInput
                                    name={"oppholdsland"}
                                    label={t("omSoekeren.oppgiLand")}
                                    rules={{pattern: /^[\w|\s]+$/}}
                                />
                            </SkjemaGruppe>

                            <RHFToValgRadio
                                name={"medlemFolketrygdenUtland"}
                                legend={t("omSoekeren.medlemFolketrygdenUtland")}
                            />
                        </>
                    )}


                    {skalSjekkeFlyktningStatus && (
                        <SkjemaGruppe>
                            <RHFToValgRadio
                                name={"flyktning"}
                                legend={<>Har du status som flyktning? <i>(gjelder alle født før 1960)</i></>}
                            />
                        </SkjemaGruppe>
                    )}

                    <br />

                    <Feilmeldinger errors={errors} />

                    <SkjemaGruppe className={"navigasjon-rad"}>
                        <Hovedknapp htmlType={"button"} onClick={handleSubmit(lagre)}>
                            {t("knapp.neste")}
                        </Hovedknapp>
                    </SkjemaGruppe>
                </form>
            </FormProvider>
        </>
    );
};

export default OmDeg;
