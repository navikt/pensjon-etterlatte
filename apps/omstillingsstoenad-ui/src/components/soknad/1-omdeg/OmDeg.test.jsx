import { render, fireEvent } from "@testing-library/react";
import OmDeg from "./OmDeg";
import { describe, expect, it, vi } from 'vitest'
import {BrowserRouter} from "react-router-dom";

vi.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    ...vi.importActual("react-i18next"),
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

// Her må det mockes litt forskjellige guid-pakker...
vi.mock("uuid", () => ({
    v4: () => "456"
}))

describe("Om deg og avdød", () => {
    it("Snapshot", () => {
        const { container, getByText } = render(<BrowserRouter><OmDeg /></BrowserRouter>);
        
        const bankkontoTypeUtenlandsk = container.querySelectorAll("[value='bankkontoType.utenlandsk']")[0];
        fireEvent.click(bankkontoTypeUtenlandsk);

        expect(getByText("omDeg.utbetalingsInformasjon.iban")).toBeDefined();
        expect(container).toMatchSnapshot();
    });
});
