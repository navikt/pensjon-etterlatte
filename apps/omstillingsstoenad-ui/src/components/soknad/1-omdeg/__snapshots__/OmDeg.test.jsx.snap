// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Om deg og avdød > Snapshot 1`] = `
<div>
  <div
    class="navds-r-mb navds-box"
    style="--__ac-r-mb-xs: var(--a-spacing-4);"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      omDeg.tittel
    </h1>
  </div>
  <form>
    <div
      class="navds-r-mb navds-box"
      style="--__ac-r-mb-xs: var(--a-spacing-4);"
    >
      <div
        class="navds-r-mb navds-box"
        style="--__ac-r-mb-xs: var(--a-spacing-0) var(--a-spacing-12);"
      >
        <div>
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="alternativAdresse"
            >
              omDeg.alternativAdresse
            </label>
            <input
              class="navds-text-field__input navds-body-short navds-body-short--medium"
              id="alternativAdresse"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-«r0»"
            />
          </div>
        </div>
      </div>
      <div
        class="navds-r-mb navds-box"
        style="--__ac-r-mb-xs: var(--a-spacing-4);"
      >
        <div
          class="navds-form-field navds-form-field--medium"
        >
          <label
            class="navds-form-field__label navds-label"
            for="kontaktinfo.telefonnummer"
          >
            omDeg.kontaktinfo.telefonnummer
          </label>
          <input
            autocomplete="tel"
            class="navds-text-field__input navds-body-short navds-body-short--medium"
            id="kontaktinfo.telefonnummer"
            inputmode="tel"
            size="25"
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-form-field__error"
            id="textField-error-«r1»"
          />
        </div>
      </div>
    </div>
    <div
      class="navds-r-mb navds-box"
      style="--__ac-r-mb-xs: var(--a-spacing-0) var(--a-spacing-12);"
    >
      <div
        class="navds-r-mb navds-box"
        style="--__ac-r-mb-xs: var(--a-spacing-4);"
      >
        <div
          id="utbetalingsInformasjon.bankkontoType"
        >
          <fieldset
            aria-labelledby="«r3»"
            class="navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
          >
            <legend
              class="navds-fieldset__legend navds-label"
              id="«r3»"
            >
              omDeg.utbetalingsInformasjon.bankkontoType
            </legend>
            <div
              class="navds-radio-buttons"
            >
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  class="navds-radio__input"
                  id="radio-«r5»"
                  name="utbetalingsInformasjon.bankkontoType"
                  type="radio"
                  value="bankkontoType.norsk"
                />
                <label
                  class="navds-radio__label"
                  for="radio-«r5»"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short navds-body-short--medium"
                    >
                      bankkontoType.norsk
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  class="navds-radio__input"
                  id="radio-«r6»"
                  name="utbetalingsInformasjon.bankkontoType"
                  type="radio"
                  value="bankkontoType.utenlandsk"
                />
                <label
                  class="navds-radio__label"
                  for="radio-«r6»"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short navds-body-short--medium"
                    >
                      bankkontoType.utenlandsk
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-fieldset__error"
              id="fieldset-error-«r4»"
            />
          </fieldset>
        </div>
      </div>
      <div
        class="navds-r-mb navds-box"
        style="--__ac-r-mb-xs: var(--a-spacing-4);"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDeg.utbetalingsInformasjon.tittel
        </h1>
      </div>
      <div
        class="navds-r-mb navds-box"
        style="--__ac-r-mb-xs: var(--a-spacing-4);"
      >
        <div>
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="utbetalingsInformasjon.utenlandskBankNavn"
            >
              omDeg.utbetalingsInformasjon.utenlandskBankNavn
            </label>
            <input
              autocomplete="off"
              class="navds-text-field__input navds-body-short navds-body-short--medium"
              id="utbetalingsInformasjon.utenlandskBankNavn"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-«ra»"
            />
          </div>
        </div>
      </div>
      <div
        class="navds-r-mb navds-box"
        style="--__ac-r-mb-xs: var(--a-spacing-4);"
      >
        <div>
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="utbetalingsInformasjon.utenlandskBankAdresse"
            >
              omDeg.utbetalingsInformasjon.utenlandskBankAdresse
            </label>
            <input
              autocomplete="off"
              class="navds-text-field__input navds-body-short navds-body-short--medium"
              id="utbetalingsInformasjon.utenlandskBankAdresse"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-«rb»"
            />
          </div>
        </div>
      </div>
      <div
        class="navds-r-mb navds-box"
        style="--__ac-r-mb-xs: var(--a-spacing-4);"
      >
        <div
          class="navds-form-field navds-form-field--medium"
        >
          <label
            class="navds-form-field__label navds-label"
            for="utbetalingsInformasjon.iban"
          >
            <div
              class="navds-stack navds-hstack navds-stack-align navds-stack-direction navds-stack-wrap"
              style="--__ac-stack-direction-xs: row; --__ac-stack-align-xs: center;"
            >
              omDeg.utbetalingsInformasjon.iban
               
              <div
                class="navds-help-text"
                data-color="info"
              >
                <button
                  aria-expanded="false"
                  class="navds-help-text__button"
                  type="button"
                >
                  <svg
                    aria-labelledby="title-«rd»"
                    class="navds-help-text__icon"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-«rd»"
                    >
                      Mer informasjon
                    </title>
                    <circle
                      cx="12"
                      cy="12"
                      fill="transparent"
                      r="11"
                      stroke="currentColor"
                      stroke-width="1.5"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <svg
                    aria-labelledby="title-«re»"
                    class="navds-help-text__icon navds-help-text__icon--filled"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-«re»"
                    >
                      Mer informasjon
                    </title>
                    <circle
                      cx="12"
                      cy="12"
                      fill="currentColor"
                      r="11"
                      stroke="currentColor"
                      stroke-width="1.5"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                      fill="var(--ax-text-accent-contrast, var(--a-surface-default))"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <div
                  aria-hidden="true"
                  class="navds-popover navds-help-text__popover navds-popover--hidden"
                  data-index="0"
                  data-placement="top"
                  style="position: absolute; left: 0px; top: 0px; transform: translate(0px, 0px);"
                >
                  <div
                    class="navds-popover__content navds-body-short"
                  >
                    omDeg.utbetalingsInformasjon.ibanHjelpetekst
                  </div>
                  <div
                    class="navds-popover__arrow"
                    style="bottom: -0.5rem;"
                  />
                </div>
              </div>
            </div>
          </label>
          <input
            autocomplete="off"
            class="navds-text-field__input navds-body-short navds-body-short--medium"
            id="utbetalingsInformasjon.iban"
            size="35"
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-form-field__error"
            id="textField-error-«rc»"
          />
        </div>
      </div>
      <div
        class="navds-r-mb navds-box"
        style="--__ac-r-mb-xs: var(--a-spacing-4);"
      >
        <div
          class="navds-form-field navds-form-field--medium"
        >
          <label
            class="navds-form-field__label navds-label"
            for="utbetalingsInformasjon.swift"
          >
            <div
              class="navds-stack navds-hstack navds-stack-align navds-stack-direction navds-stack-wrap"
              style="--__ac-stack-direction-xs: row; --__ac-stack-align-xs: center;"
            >
              omDeg.utbetalingsInformasjon.swift
               
              <div
                class="navds-help-text"
                data-color="info"
              >
                <button
                  aria-expanded="false"
                  class="navds-help-text__button"
                  type="button"
                >
                  <svg
                    aria-labelledby="title-«rh»"
                    class="navds-help-text__icon"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-«rh»"
                    >
                      Mer informasjon
                    </title>
                    <circle
                      cx="12"
                      cy="12"
                      fill="transparent"
                      r="11"
                      stroke="currentColor"
                      stroke-width="1.5"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <svg
                    aria-labelledby="title-«ri»"
                    class="navds-help-text__icon navds-help-text__icon--filled"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-«ri»"
                    >
                      Mer informasjon
                    </title>
                    <circle
                      cx="12"
                      cy="12"
                      fill="currentColor"
                      r="11"
                      stroke="currentColor"
                      stroke-width="1.5"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                      fill="var(--ax-text-accent-contrast, var(--a-surface-default))"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <div
                  aria-hidden="true"
                  class="navds-popover navds-help-text__popover navds-popover--hidden"
                  data-index="0"
                  data-placement="top"
                  style="position: absolute; left: 0px; top: 0px; transform: translate(0px, 0px);"
                >
                  <div
                    class="navds-popover__content navds-body-short"
                  >
                    omDeg.utbetalingsInformasjon.swiftHjelpetekst
                  </div>
                  <div
                    class="navds-popover__arrow"
                    style="bottom: -0.5rem;"
                  />
                </div>
              </div>
            </div>
          </label>
          <input
            autocomplete="off"
            class="navds-text-field__input navds-body-short navds-body-short--medium"
            id="utbetalingsInformasjon.swift"
            size="25"
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-form-field__error"
            id="textField-error-«rg»"
          />
        </div>
      </div>
    </div>
    <br />
    <div
      class="navds-r-mb navds-box"
      style="--__ac-r-mb-xs: var(--a-spacing-0) var(--a-spacing-12);"
    >
      <div
        class="navds-r-mb navds-stack navds-hstack navds-stack-gap navds-stack-justify navds-stack-direction navds-stack-wrap"
        style="--__ac-r-mb-xs: var(--a-spacing-0) var(--a-spacing-8); --__ac-stack-gap-xs: var(--a-spacing-4); --__ac-stack-direction-xs: row; --__ac-stack-justify-xs: center;"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          data-variant="primary"
          type="button"
        >
          <span
            class="navds-label"
          >
            knapp.neste
          </span>
        </button>
      </div>
    </div>
    <div
      class="navds-r-mb navds-stack navds-hstack navds-stack-justify navds-stack-direction navds-stack-wrap"
      style="--__ac-r-mb-xs: var(--a-spacing-0) var(--a-spacing-8); --__ac-stack-direction-xs: row; --__ac-stack-justify-xs: center;"
    >
      <button
        class="navds-button navds-button--tertiary navds-button--medium"
        data-variant="tertiary"
        id="avbryt-btn"
        type="button"
      >
        <span
          class="navds-label"
        >
          knapp.avbryt
        </span>
      </button>
    </div>
    <dialog
      aria-label="avbrytModal.spoersmaal"
      class="navds-modal navds-modal--polyfilled navds-modal--autowidth"
      role="dialog"
    >
      <div
        class="navds-modal__header"
      >
        <button
          class="navds-modal__button navds-button navds-button--tertiary-neutral navds-button--small navds-button--icon-only"
          data-color="neutral"
          data-variant="tertiary"
          type="button"
        >
          <span
            class="navds-button__icon"
          >
            <svg
              aria-labelledby="title-«r9»"
              fill="none"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title-«r9»"
              >
                Lukk
              </title>
              <path
                d="M6.53 5.47a.75.75 0 0 0-1.06 1.06L10.94 12l-5.47 5.47a.75.75 0 1 0 1.06 1.06L12 13.06l5.47 5.47a.75.75 0 1 0 1.06-1.06L13.06 12l5.47-5.47a.75.75 0 0 0-1.06-1.06L12 10.94z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
        <h1
          class="navds-heading navds-heading--medium"
        >
          avbrytModal.spoersmaal
        </h1>
      </div>
      <div
        class="navds-modal__body"
      >
        <div
          class="navds-r-mi navds-stack navds-vstack navds-stack-gap navds-stack-direction"
          style="--__ac-r-mi-xs: var(--a-spacing-8); --__ac-stack-gap-xs: var(--a-spacing-12); --__ac-stack-direction-xs: column;"
        >
          <p
            class="mute avbryt-text navds-body-short navds-body-short--medium"
          >
            avbrytModal.informasjon
          </p>
          <div
            class="navds-stack navds-hstack navds-stack-gap navds-stack-justify navds-stack-direction navds-stack-wrap"
            style="--__ac-stack-gap-xs: var(--a-spacing-4); --__ac-stack-direction-xs: row; --__ac-stack-justify-xs: center;"
          >
            <button
              class="navds-button navds-button--secondary navds-button--medium"
              data-variant="secondary"
              id="avbryt-nei-btn"
              type="button"
            >
              <span
                class="navds-label"
              >
                avbrytModal.svarNei
              </span>
            </button>
            <button
              class="navds-button navds-button--primary navds-button--medium"
              data-variant="primary"
              id="avbryt-ja-btn"
              type="button"
            >
              <span
                class="navds-label"
              >
                avbrytModal.svarJa
              </span>
            </button>
          </div>
          <div
            class="navds-stack navds-hstack navds-stack-justify navds-stack-direction navds-stack-wrap"
            style="--__ac-stack-direction-xs: row; --__ac-stack-justify-xs: center;"
          >
            <button
              class="navds-button navds-button--tertiary navds-button--medium"
              data-variant="tertiary"
              id="slett-soeknad"
              style="color: var(--a-surface-danger);"
              type="button"
            >
              <span
                class="navds-label"
              >
                avbrytModal.svarSlett
              </span>
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </form>
</div>
`;
